app/model/Location.scala

```diff
@@ -41,16 +42,24 @@ case class Location(name: String, url: String, queryParams: Map[String, String]
 
 object Locations {
   val personalTaxAccountLocationName = "personal-tax-account"
-  val businessTaxAccountLocationName = "business-tax-account"
-  val taxAccountRouterHomeLocationName = "tax-account-router"
-  lazy val PersonalTaxAccount = Location(personalTaxAccountLocationName, ExternalUrls.getUrl(personalTaxAccountLocationName))
-  lazy val BusinessTaxAccount = Location(businessTaxAccountLocationName, ExternalUrls.getUrl(businessTaxAccountLocationName))
-  lazy val TaxAccountRouterHome = Location(taxAccountRouterHomeLocationName, s"${ExternalUrls.taxAccountRouterHost}/account")
 
+  lazy val PersonalTaxAccount = Location(personalTaxAccountLocationName, ExternalUrls.getUrl(personalTaxAccountLocationName))
+  lazy val BusinessTaxAccount = locationFromConf("bta")
+  lazy val TaxAccountRouterHome = locationFromConf("tax-account-router")
   val twoStepVerificationLocationName = "two-step-verification"
   val twoStepVerificationRequiredLocationName = "two-step-verification-required"
 
-  def twoStepVerification(queryString: Map[String, String]) = Location(twoStepVerificationLocationName, ExternalUrls.getUrl(twoStepVerificationLocationName), queryString)
+  def locationFromConf(location: String) = configuration.getConfig(s"locations.$location").map { conf =>
```

Comment:
we could add a getConfig method that throws an exception with the appropriate error message if a config does not exist:
something like:
``` scala 
  private def getConfig(key: String) = current.configuration.getString(key)
    .getOrElse(throw new RuntimeException(s"[$key] is not configured!"))
```

if you need it in a generic way you could add:
```scala
  private def getConfig[T](key: String, block: String => Option[T]) = block(key)
    .getOrElse(throw new RuntimeException(s"[$key] is not configured!"))
```